fuzzing/private/oss_fuzz/package.bzl

```diff
@@ -25,6 +25,28 @@ def _oss_fuzz_package_impl(ctx):
         action_inputs.append(binary_info.corpus_dir)
     if binary_info.dictionary_file:
         action_inputs.append(binary_info.dictionary_file)
+
+    binary_deps = [
+        runfile
+        for runfile in binary_info.binary_runfiles.files.to_list()
+        if runfile != binary_info.binary_file
+    ]
+    action_inputs += binary_deps
+    binary_deps_dir = ctx.attr.base_name + "_deps"
```

Comment:
I believe the Bazel convention is to have the runfiles of a binary `<name>` as a separate directory `<name>.runfiles` side-by-side with the binary (e.g., as documented [here](https://docs.bazel.build/versions/master/skylark/rules.html#runfiles-location)).

Should we preserve this convention here, too? This way, programs that are aware of their runfiles will be able to automatically discover this directory based on this convention.

Reply:
That makes sense, I will change it. 

We will deviate from what Bazel does in that we currently don't mimic the full runfile structure under `<name>.runfiles`, but only a flat structure where all runfile paths are truncated to their base names. Is that okay or should I extend the logic here?

Reply:
> That makes sense, I will change it.
> 
> We will deviate from what Bazel does in that we currently don't mimic the full runfile structure under `<name>.runfiles`, but only a flat structure where all runfile paths are truncated to their base names. Is that okay or should I extend the logic here?

I think that would cause issues, since the runfiles structure is a contract that Bazel provides to the binaries that have data dependencies. Moreover, what happens for distinct paths that share the same base name? I do think we should preserve the actual runfile structure in the OSS-Fuzz packaging.

Reply:
While it's possible to port the entire runfiles tree to OSS-Fuzz, I think that we would need to go pretty deep here. Basically this requires:
1. Looping over all `short_path`s computing the longest prefix chain of `../`.
2. Moving the actual binary that deep into a synthetic folder structure.
3. Move it's runfiles relative to it based on their `short_path`s, creating directories as needed.
4. Emit a top-level wrapper binary that `cd`s to the binary and executes it.

I can do that and would find it useful, but I'm not sure I'll be able to pull it off in pure bash. Should I start with a mix of bash/Starlark and we clean it up later?

Reply:
This may not be that difficult after all. `short_path` always starts with at most one `../` (in case of a runfiles in another workspace), right?